### PR TITLE
Issues/1631 autoscaling docs (resolves #1631)

### LINKED ISSUE #1631: Improve autoscaling documentation
The autoscaling documentation would benefit from a list of steps for getting started. Also many terms such as "Toil Appliance" could be better defined and use links to other parts of documentation.

### PR DESCRIPTION
**Pull request by jessebrennan**
This PR contains fixes to several preexisting build errors in the docs.

It also drastically reorganized the autoscaling docs, and the cluster utilities. The most significant updates are in commits. 

* 	bd8ce1d (Clarify install docs, remove CGCloud instructions)
* 	5cb2bad (Clarify command line options)
* 	6e5976f (Update autoscaling instructions)

I've read over it lots of times but there are surely typos that I have missed. If you see anything I've left out (or god forbid) explained incorrectly about using autoscaling please review!

Jenkins, skip this please

### CODE REVIEW

**Comment by jessebrennan** [comment]
Jenkins skip this please!

File: docs/cli.rst [hunk 1]
```diff
@@ -137,5 +138,86 @@ such a deadlock can not occur.
...
+   Boto must be `configured`_ with AWS credentials before using cluster utilities.
+
+.. _configured: http://boto3.readthedocs.io/en/latest/guide/quickstart.html#configuration
+
+.. _launchCluster:
+
+launch-cluster
+^^^^^^^^^^^^^^
+
+Running ``toil launch cluster`` starts up a leader for a cluster. Workers can be
```
Comment (cket) [inline docs/cli.rst]: launch-cluster

File: docs/cli.rst [hunk 2]
```diff
@@ -137,5 +138,86 @@ such a deadlock can not occur.
...
+addee to the initial cluster by specifying the ``-w`` option. For an example usage see
```
Comment (cket) [inline docs/cli.rst]: added

File: docs/cli.rst [hunk 3]
```diff
@@ -137,5 +138,86 @@ such a deadlock can not occur.
...
+:ref:`launchingCluster`. More information can be found using the ``--help`` option.
+
+.. _sshCluster:
+
+ssh-cluster
+^^^^^^^^^^^
+
+Toil provides the ability to ssh into the leader of the cluster.
+
+The last utility provided by the Toil Provisioner is ``ssh-cluster`` and it
```
Comment (cket) [inline docs/cli.rst]: Can you reword this? It's not the last utility since rsync is below this

**Comment by jessebrennan** [comment]
Jenkins skip this please!

**Comment by jessebrennan** [comment]
@cket This could probably be merged as is, but instead I'm going to add in more on static provisioning.

**Comment by jessebrennan** [comment]
@ejacox I think this is ready.

**Pull request merged**